google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/dialect/InterleavedTest.java

```diff
@@ -0,0 +1,73 @@
+package org.hibernate.dialect;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.testing.transaction.TransactionUtil.doInJPA;
+
+import java.util.List;
+import java.util.Map;
+import javax.persistence.EntityManager;
+import org.hibernate.Session;
+import org.hibernate.dialect.entities.Child;
+import org.hibernate.dialect.entities.Child.ChildId;
+import org.hibernate.dialect.entities.GrandParent;
+import org.hibernate.dialect.entities.Parent;
+import org.hibernate.dialect.entities.Parent.ParentId;
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+import org.junit.Test;
+
+public class InterleavedTest extends BaseEntityManagerFunctionalTestCase {
+
+  @Override
+  protected Class<?>[] getAnnotatedClasses() {
+    return new Class<?>[]{
+        Parent.class,
+        GrandParent.class,
+        Child.class
+    };
+  }
+
+  /**
+   * We override the default with the 'create-drop' mode because we would like to verify that the
+   * interleaved tables are created and dropped in the correct order:
+   * parents created before children; children dropped before parents.
+   */
+  @Override
+  protected Map buildSettings() {
+    Map settings = super.buildSettings();
+    settings.put(org.hibernate.cfg.AvailableSettings.HBM2DDL_AUTO, "create-drop");
+    return settings;
+  }
+
+  @Test
+  public void testInterleavedEntities() {
```

Comment:
How is this specifically testing `@Interleaved` though? In other words, how would this break if we didn't support interleaved?

Do you need to verify schema generation that it includes the `INTERLEAVED` keyword or something?

Reply:
Done. Ah good point. I added in the test to now check the database schema to verify that the tables are interleaved with the correct parent.

This was done by querying the `information_schema.tables` table and checking the `parent_table_name` field is set correctly for interleaved tables. See: https://cloud.google.com/spanner/docs/information-schema#tables